cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala

```diff
@@ -19,26 +19,51 @@
  */
 package org.neo4j.cypher.docgen
 
-import org.junit.Test
-import org.junit.Assert._
 import org.neo4j.cypher.QueryStatisticsTestSupport
-import org.neo4j.visualization.graphviz.GraphStyle
-import org.neo4j.visualization.graphviz.AsciiDocSimpleStyle
+import org.neo4j.cypher.docgen.tooling.{DocBuilder, DocumentingTest, ResultAssertions}
 
-class ForeachTest extends DocumentingTestBase with QueryStatisticsTestSupport with SoftReset {
-  override def graphDescription = List("A KNOWS B", "B KNOWS C", "C KNOWS D")
+class ForeachTest extends DocumentingTest with QueryStatisticsTestSupport {
+  override def outputPath = "target/docs/dev/ql"
 
-  override protected def getGraphvizStyle: GraphStyle =
-    AsciiDocSimpleStyle.withAutomaticRelationshipTypeColors()
+  override def doc = new DocBuilder {
+    doc("FOREACH", "query-foreach")
+    initQueries("""CREATE (a:Person {name: 'A'}),
+                  |       (b:Person {name: 'B'}),
+                  |       (c:Person {name: 'C'}),
+                  |       (d:Person {name: 'D'}),
+                  |       (a)-[:KNOWS]->(b),
+                  |       (b)-[:KNOWS]->(c),
+                  |       (c)-[:KNOWS]->(d)""")
+    synopsis("The `FOREACH` clause is used to update data within a list, whether components of a path, or result of aggregation.")
+    p(
+      """* <<query-foreach-introduction, Introduction>>
+        |* <<foreach-mark-all-nodes-along-a-path, Mark all nodes along a path>>
+      """.stripMargin)
+    section("Introduction", "query-foreach-introduction") {
+      p(
+        """Lists and paths are key concepts in Cypher.
+          |`FOREACH` can be used to update data, such as executing update commands on elements in a path, or on a list created by aggregation.""".stripMargin)
+      p("""The variable context within the `FOREACH` parenthesis is separate from the one outside it.
+          |This means that if you `CREATE` a node variable within a `FOREACH`, you will _not_ be able to use it outside of the foreach statement, unless you match to find it.""".stripMargin)
+      p(
+        """Within the `FOREACH` parentheses, you can do any of the updating commands -- `CREATE`, `CREATE UNIQUE`, `MERGE`, `DELETE`, and `FOREACH`.""".stripMargin)
+      p("""If you want to execute an additional `MATCH` for each element in a list then `UNWIND` (see <<query-unwind>>) would be a more appropriate command.""".stripMargin)
+      graphViz()
```

Comment:
@petraselmer The image here (and in one more place) does not render when I build. Is there something I am missing?

Reply:
@mariascharin hope all is resolved here? Let me know if not

Reply:
@petraselmer No we haven't been able to figure it out yet.